lib/Alchemy/Phrasea/PhraseanetService/Controller/PSExposeController.php

```diff
@@ -711,6 +713,75 @@ public function addPublicationAssetsAction(PhraseaApplication $app, Request $req
         ]);
     }
 
+    /**
+     * @param PhraseaApplication $app
+     * @param Request $request
+     * @return string
+     */
+    public function getDataboxesFieldAction(PhraseaApplication $app, Request $request)
+    {
+        $exposeName = $request->get('exposeName');
+        $databoxes = $this->getApplicationBox()->get_databoxes();
+
+        $fields = [];
+        foreach ($databoxes as $databox) {
+            foreach ($databox->get_meta_structure() as $meta) {
+                // get databoxID_metaID for the checkbox name
+                $fields[$databox->get_viewname()][$meta->get_id()]['id'] = $databox->get_sbas_id().'_'.$meta->get_id();
+                $fields[$databox->get_viewname()][$meta->get_id()]['name'] = $meta->get_label($this->app['locale']);
+            }
+        }
+
+        $actualFieldList = [];
+        if (!empty($exposeName)) {
+            /** @var Expose $ex */
+            $ex = $this->app['ps_settings.expose'];
+            $exposeInstance = $ex->getInstance($exposeName);
+
+            $actualFieldList = json_decode($exposeInstance->getFieldList(), true);
+        }
+
+        return $this->render('prod/WorkZone/ExposeFieldMapping.html.twig', [
+            'fields'            => $fields,
+            'exposeName'        => $exposeName,
+            'actualFieldList'   => $actualFieldList
+        ]);
+    }
+
+    /**
+     * @param PhraseaApplication $app
+     * @param Request $request
+     * @return \Symfony\Component\HttpFoundation\JsonResponse
+     */
+    public function saveFieldMappingAction(PhraseaApplication $app, Request $request)
+    {
+        $exposeName = $request->get('exposeName');
+        $fields = ($request->request->get('fields') === null) ? null : json_encode(array_keys($request->request->get('fields')));
+
```

Comment:
too easy !
the goal is to save settings to an extensible format.
If tomorrow we want to add attributes to a field (eg. a simple color ore a complex acl), event in this json it's not possible because it's a simple list)
we should
- save each field as a specific setting (child of a common "fields" parent), with fieldname/or/int as key.
or
- save all fields as json, but with each field as an (extensible) object